lib/SIL/Verifier/SILVerifier.cpp

```diff
@@ -5038,6 +5120,12 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
             require(state.CFG == foundState.CFG,
                     "inconsistent coroutine states entering basic block");
```

Comment:
You need to add something here to verify that we have the same `GotAsyncContinuation` on all predecessors.